INCR-209 move python 2 to 3 djangoapps user_api accounts

jainmickey opened the pull request:
INCR-209 Run python-modernize on openedx/core/djangoapps/user_api/accounts

edx-webhook commented:
Thanks for the pull request, @jainmickey! I've created [OSPR-3379](https://openedx.atlassian.net/browse/OSPR-3379) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

natabene commented:
@jainmickey Thank you for your contribution. Please let us know once all checks are green.

michaelyoungstrom commented:
jenkins run all

jainmickey commented:
@jmbowman `openedx/core/djangoapps/user_api/accounts/tests/test_retirement_views.py:1336` here we are trying to patch `is_username_retired` in user_api/accounts/views but it doesn't exists there anymore.
Did we removed it or we replaced with something?

jmbowman commented:
`is_username_retired` is imported in user_api/accounts/views: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/accounts/views.py#L66 , but this PR removed it for some reason.  Should just need to restore that import to fix the failing test.

It also looks like you'll need to rebase from master and resolve a conflict in one recently changed file before we can merge this.

The causes of the quality job failure can be found here: https://build.testeng.edx.org/job/edx-platform-quality-pipeline-pr/3637/Diff_20Quality_20Report/ .  Relevant tips at https://openedx.atlassian.net/wiki/spaces/AC/pages/977666218/Using+python-modernize+effectively

jainmickey commented on openedx/core/djangoapps/user_api/accounts/utils.py:
```diff
@@ -37,10 +38,12 @@ def validate_social_link(platform_name, new_social_link):
+                                      'username or a URL that contains')
+        translated_error_message2 = _('To remove the link from your edX profile,'
+                                      ' leave this field blank.')
+        raise ValueError(' ' + translated_error_message1 + ' "' + required_url_stub +
```
@jmbowman I was getting an error from `pylint` as:
`i18n function _() must be called with a literal string (translation-of-non-string)`

So, I changed it in this way. Let me know if this is correct or not?

jmbowman replied:
It's good that you caught this, since only the first part of that text made it into the translation files, which has apparently led to translations mentioning the need to use a URL containing a double quote character.  But I think an even better fix would be to use a placeholder in the translated literal, which will be replaced at runtime with the correct URL stub.  Please read our [docs on creating template strings for translation](https://edx.readthedocs.io/projects/edx-developer-guide/en/latest/internationalization/i18n.html#do-not-append-strings-or-interpolate-values).

This will require all the translations of this message to be updated, but that should be ok in this case since they're incorrect anyway.

jainmickey replied:
That looks good, I updated the code to match it. Let me know if it requires any other change as well!

jainmickey commented:
@jmbowman thanks for helping in this out, added the changes!!

natabene commented:
jenkins run all

jmbowman commented:
jenkins run all

jmbowman commented:
jenkins run bokchoy

edx-webhook commented:
@jainmickey 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-platform&entry.867055334=https://github.com/edx/edx-platform/pull/20436&entry.1484655318=https://github.com/jainmickey&entry.752974735=2019-05-06+21:06&entry.1917517419=2019-05-08+20:26&entry.2133058324=Yes) so we can improve your experience in the future.
